packages/app/src/state/cortex-config/types.ts

```diff
@@ -0,0 +1,287 @@
+/**
+ * Copyright 2021 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as yup from "yup";
+
+const isDuration = {
+  name: "is-duration",
+  message: "Must be a valid duration: [number][s|m|h|d|w], e.g. 2d for 2 days.",
+  test: (value: any) =>
+    typeof value === "undefined" || /^\d+[s|m|h|d|w]{1}$/.test(value || "") // allow this field to be undefined or be a valid duration
+};
+
+export const cortexLimitsSchema = yup.object({
+  ingestion_rate: yup.number().integer().min(0, "Cannot be negative").meta({
+    description: "Ingestion rate limit in samples per second for this tenant."
```

Comment:
For making the meaning of this parameter unambiguous, we need to inform about the rate limit strategy being `local`:

> f.StringVar(&l.IngestionRateStrategy, "distributor.ingestion-rate-limit-strategy", "local", "Whether the ingestion rate limit should be applied individually to each distributor instance (local), or evenly shared across the cluster (global).")

We explicitly choose this to be `local`, see 

https://github.com/opstrace/opstrace/blob/17b792ddf72e29f941a33b6e0bba0acaf5c58874/packages/controller/src/resources/cortex/index.ts#L305

I.e. this is a per-ingester limit.

Reply:
Yes, good point, thanks!